File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        if n > 1 and nums[0] == 0:
            return False
    
        target_index = n - 1
        i = n - 2

        for i in reversed(range(n-1)):
            if nums[i] >= target_index - i:
                target_index = i



        return target_index == 0

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int pointer = 1;

        for (int i=1; i < nums.size(); i++) {
            if (nums[i] != nums[i-1]) {
                nums[pointer] = nums[i];
                pointer++;
            }
        }

        return pointer;
    }
};